src/sagemaker/amazon/factorization_machines.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from sagemaker.amazon.amazon_estimator import AmazonAlgorithmEstimatorBase, registry
+from sagemaker.amazon.common import numpy_to_record_serializer, record_deserializer
+from sagemaker.amazon.hyperparameter import Hyperparameter as hp  # noqa
+from sagemaker.amazon.validation import gt, isin, isint, ge, isfloat
+from sagemaker.predictor import RealTimePredictor
+from sagemaker.model import Model
+from sagemaker.session import Session
+
+
+class FactorizationMachines(AmazonAlgorithmEstimatorBase):
+
+    repo = 'factorization-machines:1'
+
+    num_factors = hp('num_factors', (gt(0), isint), 'An integer greater than zero')
+    predictor_type = hp('predictor_type', isin('binary_classifier', 'regressor'),
+                        'Value "binary_classifier" or "regressor"')
+    epochs = hp('epochs', (gt(0), isint), "An integer greater than 0")
+    clip_gradient = hp('clip_gradient', isfloat, "A float value")
```

Comment:
This forces an explicit assignment, where an implicit assignment would be fine.

If I have a FacotrizationMachines object fm, then:

```
fm.clip_gradient = 55
```

will fail, because 55 is not float - even though it can be represented as a float. Consider using the ```isnumber``` check instead.

Reply:
That's a good idea - will change